modules/sigprofiler/1.0/sigprofiler.smk

```diff
@@ -0,0 +1,215 @@
+    conda: CFG["conda_envs"]["sigprofiler"]
+    threads: CFG["threads"]["generator"]
+    shell:
+        "python {input.script} {wildcards.sample_set} {params.ref} {input.maf}"
+
+rule _sigprofiler_run_estimate:
+    input:
+        unpack(get_dir)
+    output:
+        stat = CFG["dirs"]["estimate"]+"{seq_type}--{genome_build}/{sample_set}/{type}/All_solutions_stat.csv"
+    params:
+        ref = lambda w: {"grch37":"GRCh37", "hg19":"GRCh37",
+                         "grch38": "GRCh38", "hg38": "GRCh38"}[w.genome_build],
+        context_type = '96,DINUC,ID',
+        exome = lambda w: {'genome': 'False', 'capture': 'True'}[w.seq_type],
+        min_sig = 1,
+        max_sig = lambda w: max_sigs[w.type],
+        nmf_repl = 30,
+        norm = 'gmm',
+        nmf_init = 'nndsvd_min',
+        outpath = CFG["dirs"]["estimate"]+"{seq_type}--{genome_build}/{sample_set}"
+    conda: CFG["conda_envs"]["sigprofiler"]
+    threads: CFG["threads"]["estimate"]
+    shell:
+        op.as_one_line("""
+        python {input.script} {input.mat} {params.outpath} 
+        {params.ref} {params.context_type} {params.exome} {params.min_sig} {params.max_sig} 
+        {params.nmf_repl} {params.norm} {params.nmf_init} {threads}
+        """)
+
+rule _sigprofiler_run_extract:
+    input:
+        unpack(get_dir),
+        stat = str(rules._sigprofiler_run_estimate.output.stat)
+    output:
+        decomp = CFG["dirs"]["extract"]+"{seq_type}--{genome_build}/{sample_set}/{type}/Suggested_Solution/COSMIC_{type}_Decomposed_Solution/De_Novo_map_to_COSMIC_{type}.csv"
+    params:
+        ref = lambda w: {"grch37":"GRCh37", "hg19":"GRCh37", 
+                         "grch38": "GRCh38", "hg38": "GRCh38"}[w.genome_build],
+        context_type = '96,DINUC,ID',
+        exome = lambda w: {'genome': 'False', 'capture': 'True'}[w.seq_type],
+        nmf_repl = 200,
+        norm = 'gmm',
+        nmf_init = 'nndsvd_min',
+        outpath = CFG["dirs"]["extract"]+"{seq_type}--{genome_build}/{sample_set}"
+    conda: CFG["conda_envs"]["sigprofiler"]
+    threads: CFG["threads"]["extract"]
+    shell:
+        op.as_one_line("""
+        python {input.script} {input.mat} {params.outpath} 
+        {params.ref} {params.context_type} {params.exome}
+        $(awk 'BEGIN {{OFS=FS=","}} $1 ~ "*" {{S=substr($1,1,length($1)-1); if (S-2<1) {{print 1}} else {{print S-2}}}}' {input.stat})
+        $(awk 'BEGIN {{OFS=FS=","}} $1 ~ "*" {{S=substr($1,1,length($1)-1); print S+2}}' {input.stat})
+        {params.nmf_repl} {params.norm} {params.nmf_init} {threads}
+        """)
+
+# Symlinks the final output files into the module results directory (under '99-outputs/')
+rule _sigprofiler_output_tsv:
+    input:
+        decomp = str(rules._sigprofiler_run_extract.output.decomp)
+    output:
+        decomp = CFG["dirs"]["outputs"] + "cosmic_sigs/{seq_type}--{genome_build}/{sample_set}.COSMIC_{type}.De_Novo_map_to_COSMIC_{type}.csv"
```

Comment:
Is genome_build here still actually genome_build or is it the projection?

Reply:
This refers to the projection build. See the expand call in rule `_sigprofiler_all`